### DISCUSSION TITLE
Problem on slot effect number 2

### DISCUSSION THREAD

**Discussion by curtistornade** [demberto/PyFLP#175] [Q&A]
Hi !
Recently, I wanted to get all the plugin name on one channel in the mixer but I ran into a problem because it appears that the slot number 2 is not showing up correctly. 
In fact, to test this error, I have put 9 Fruity Parametric EQ 2 on one channel (every slot except the third one) and one Fruity Reeverb 2 on the third slot of this channel. 

![Capture d’écran 2023-08-26 à 17 09 17](https://github.com/demberto/PyFLP/assets/91035716/b9cfb2d5-b37e-4c35-9421-035c7917218d)

Next, I ran this following script : 

```
project = pyflp.parse(path)
    list_all_effects = []
    for insert in range(project.mixer.__len__() - 1):
        for slot in range(project.mixer.max_slots):
            if project.mixer.__getitem__(insert).__getitem__(slot).plugin is not None:
                list_all_effects.append(project.mixer.__getitem__(insert).__getitem__(slot))
                print('SLOT : ', slot+1)
                print('Effect : ', project.mixer.__getitem__(insert).__getitem__(slot) .internal_name)
```

And this is what I got : 

```
SLOT 1
Effect :  Fruity Parametric EQ 2
SLOT 2
Effect :  Fruity Reeverb 2
SLOT 3
Effect :  Fruity Parametric EQ 2
SLOT 4
Effect :  Fruity Parametric EQ 2
SLOT 5
Effect :  Fruity Parametric EQ 2
SLOT 6
Effect :  Fruity Parametric EQ 2
SLOT 7
Effect :  Fruity Parametric EQ 2
SLOT 8
Effect :  Fruity Parametric EQ 2
SLOT 9
Effect :  Fruity Parametric EQ 2
```

As we can see, the Fruity Reeverb 2 is in this case showing on the second slot instead of the third one and there is no plugin detected on the last slot.
I have tested this with a Fruity Reeverb 2 on the second slot and this is the output : 

![Capture d’écran 2023-08-26 à 17 10 17](https://github.com/demberto/PyFLP/assets/91035716/3f656546-82e7-47eb-a678-d9aa376945c5)

```
SLOT :  1
Effect :  Fruity Parametric EQ 2
SLOT :  2
Effect :  Fruity Parametric EQ 2
SLOT :  3
Effect :  Fruity Parametric EQ 2
SLOT :  4
Effect :  Fruity Parametric EQ 2
SLOT :  5
Effect :  Fruity Parametric EQ 2
SLOT :  6
Effect :  Fruity Parametric EQ 2
SLOT :  7
Effect :  Fruity Parametric EQ 2
SLOT :  8
Effect :  Fruity Parametric EQ 2
SLOT :  9
Effect :  Fruity Parametric EQ 2
```

So it seems that the slot 2 is not read by the script and the following slots are shifted back by 1. Do you have any clue on this issue ?
Thank you a lot  🙏

**Comment by curtistornade** [comment]
To add some further information on this issue, I have tested the case where I insert an effect on SLOT 2 and SLOT 3 or more and there was no problem at all.
But as soon as I add something on SLOT 1, SLOT 2 is not processed by the script and it is the same problem as described before.

**Reply by demberto** [reply]
Send the FLP. Which version of PyFLP are you using? Why are you looping over the mixer and inserts with an index?

**Reply by curtistornade** [reply]
Here is the FLP file : [parser_slot_test.flp.zip](https://github.com/demberto/PyFLP/files/12448109/parser_slot_test.flp.zip)

I'm using the version 2.2.1 of PyFLP. And I'm looping over with an index because I didn't see the iterator method in the first place. But it is the same problem with the iterator too.

Thank you for your answer 🙏
